#161982646 fix login and signup errors

michael-basweti opened the pull request:
## What does this PR do?
* fix errors in login and signup

## Description of Task to be completed?
* Added logout
* Added more validations
* Made modals for login and signup routes

## How should this be manually tested?
* clone [this](https://github.com/andela/ah-shakas-frontend.git) repo.
* run command `git checkout bg-fix-login-161982646`.
* run `yarn install` to install requirements
* run `yarn start` to run project
* navigate to [http://localhost:3000/](http://localhost:3000/) locally to access the system locally or [http://10.10.162.103:3000/](http://10.10.162.103:3000/)  to access it online.

[sample pr](https://github.com/andela/ah-shakas/pull/39)

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #]

lukorito commented on src/actions/login.action.js:
```diff
@@ -5,7 +5,7 @@ const { LOGIN } = CONSTANTS;
   type: LOGIN,
   payload: api({
-    url: 'http://127.0.0.1:8000/api/users/login/',
+    url: 'https://ah-shakas-staging.herokuapp.com/api/users/login/',
```
Shouldn't this be in the base url?

Arthur236 replied:
I agree with @lukorito

lukorito commented on src/components/Auth/LoginComponent.jsx:
```diff
@@ -30,13 +32,24 @@ class LoginComponent extends Component {
         window.location.replace('/');
```
You could use redirect from `react router dom`

lukorito replied:
https://reacttraining.com/react-router/web/api/Redirect

Arthur236 replied:
There is also the option of using 
```
this.props.history.push('/')
```
Since you are not redirecting to a link outside our application, I would prefer you use either the `Redirect` component or the `history` prop

lukorito commented on src/components/Auth/LoginComponent.jsx:
```diff
@@ -30,13 +32,24 @@ class LoginComponent extends Component {
+    const err = { ...redirect.logindata };
+    const err1 = { ...err.errors };
+    const err2 = { ...err1.response };
+    const err3 = { ...err2.data };
```
Why so much redundancy here?

michael-basweti replied:
@lukorito how do you think I should do this?

Arthur236 commented on src/components/Auth/LoginComponent.jsx:
```diff
@@ -30,13 +32,24 @@ class LoginComponent extends Component {
+    const err2 = { ...err1.response };
+    const err3 = { ...err2.data };
+    const err4 = { ...err3.errors };
+    const emailerr = { ...err4.email };
```
I feel there's a better way you could handle this instead of having so many const declarations

michael-basweti replied:
I honestly would like a suggestion on how I would do this @Arthur236 @lukorito

Arthur236 commented on src/components/Dropdown.jsx:
```diff
@@ -3,12 +3,23 @@ import PropTypes from 'prop-types';
+const Logout = () => {
+  if (user) {
+    localStorage.removeItem('user');
+    window.location.replace('/');
```
Same case here.
Use either the `Redirect` component or the `history` prop

Arthur236 commented on src/components/Navbar.jsx:
```diff
@@ -1,17 +1,31 @@
-    openRegistration: false,
-    menuOpen: false,
-  };
+  constructor(props) {
```
I thought we agreed on using ES6 syntax, and as such, we should not be using the constructor method

michael-basweti replied:
@lukorito can explain this better, I will lias with him tomorrow

Arthur236 commented on src/components/Navbar.jsx:
```diff
@@ -43,18 +57,18 @@ class Navbar extends React.Component {
-              <NavItem href="#"><Icon>notifications_none</Icon></NavItem>
-              <NavItem onClick={this.toggleMenu}><Icon>more_vert</Icon></NavItem>
-            </>
+              <>
```
Instead of having empty tags, use `<React.Fragment></React.Fragment>`

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

michael-basweti committed:
bug(fix-login): fix login and signup errors  …
- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

michael-basweti committed:
bug(fix-login): fix login and signup errors

- make login work online
- add validations on both registration and login
- make the login and registration routes modals
- add logout

[Delivers #161982646]

Arthur236 reviewed (approved):
Looks good 👍

lukorito commented on src/components/Auth/LoginComponent.jsx:
```diff
@@ -22,21 +24,37 @@ class LoginComponent extends Component {
     }
+
+    if (logindata.onRejected) {
+      const { errors } = logindata.errors.response.data;
```
nice

Arthur236 commented:
@lukorito could you approve the changes you requested so I can merge this
